src/modules/dotnet.rs

```diff
@@ -39,20 +42,89 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     // Internally, this module uses its own mechanism for version detection.
     // Typically it is twice as fast as running `dotnet --version`.
     let enable_heuristic = config.heuristic;
-    let version = if enable_heuristic {
-        let repo_root = context.get_repo().ok().and_then(|r| r.root.as_deref());
-        estimate_dotnet_version(&dotnet_files, &context.current_dir, repo_root)?
+
+    let formatter = if let Ok(formatter) = StringFormatter::new(config.format) {
+        formatter
+            .map_style(|variable| match variable {
+                "style" => Some(config.style.to_string()),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let version = if enable_heuristic {
+                        let repo_root = context.get_repo().ok().and_then(|r| r.root.as_deref());
+                        estimate_dotnet_version(&dotnet_files, &context.current_dir, repo_root)?
```

Comment:
You may want to return an error (`Some(Err(StringFormatterError::new(..)))`) here.

Originally failure in finding dotnet version will make **the module** empty. Returning `None` will make **the variable** `$version` empty, while throwing error makes the format string fail to render.